### ISSUE TITLE
Enable CI

### ISSUE THREAD

**Issue by kmpaul** [NCAR/notebook-gallery#25]
It would be good to enable CircleCI (with nightly builds) on this.

@bonnland

**Comment by kmpaul** [comment]
A little investigation suggests that there is something that might work for us:

https://github.com/computationalmodelling/nbval

**Comment by kmpaul** [comment]
However, as is usually the case, the main issue will be if notebooks require Dask.  If we need parallelism, which probably also means we need to run on Cheyenne or Casper, then we need a system to support this.

**Comment by bonnland** [comment]
As an initial try, I'm borrowing heavily from the parcels repo, which uses Travis to run nbval on a set of examples: 

https://github.com/OceanParcels/parcels/blob/master/.travis.yml

Note that I've used just CircleCI in the past, so I'm not sure if Travis has any gotchas we should be worried about.  

It looks like you can sign into Travis using your GitHub account and see the build & test output I'm getting so far here: 

https://travis-ci.org/NCAR/notebook-gallery

**Comment by bonnland** [comment]
@kmpaul Any opinions on these questions:  

- how much of a priority is it that these notebooks work?   @andersy005 mentioned yesterday that these notebooks may not require Dask per se, but the presence of local data files.   This could be addressed with changes to the intake catalogs, but maybe this is not a priority right now? 
- Are nightly builds important right now, while all the notebooks are producing errors?
- What kind of email notifications, if any, should be set up?   This is turned off now in the Travis config.
- We could, without much effort, switch to CircleCI.  CISL's SWES group is managing a special account where our notebook gallery could be added, and that would give us more than the normal "free account" options for CircleCI.   Is this worth pursuing?   I think Dave Vance would be the person to talk to.

**Comment by kmpaul** [comment]
@bonnland:

> * how much of a priority is it that these notebooks work?   @andersy005 mentioned yesterday that these notebooks may not require Dask per se, but the presence of local data files.   This could be addressed with changes to the intake catalogs, but maybe this is not a priority right now?

...Well, I think we should get these notebooks to work on a CI platform, if the data can be made local.  Maybe that means we need a sample dataset in this repo and a sample intake catalog to go with it, so that we can demonstrate "proper" use.  But, in the end, I would say that _everything_ should be tested/testable on CI.

...But that could also mean that we need a CI platform deployed locally with access to GLADE.  

> * Are nightly builds important right now, while all the notebooks are producing errors?

No.  I'd make nightlies a priority only after the other notebooks are passing tests.  ...But that's only true if we can get the notebooks to pass tests fairly soon.  If the effort to get these notebooks to work takes too long, then we will need to keep iterating on this with new versions of software until it works with the latest versions.  ...And that can best be done with nightlies, so it's a mixed answer.

> * What kind of email notifications, if any, should be set up?   This is turned off now in the Travis config.

Good question.  I would keep it turned off for now and only include @jukent, me and you, until the notebooks are passing tests.  Then, once the notebooks are passing tests, I think it's fair to change the notifications to include all of @NCAR/xdev.

> * We could, without much effort, switch to CircleCI.  CISL's SWES group is managing a special account where our notebook gallery could be added, and that would give us more than the normal "free account" options for CircleCI.   Is this worth pursuing?   I think Dave Vance would be the person to talk to.

I like CircleCI.  More than I like Travis, but that's just a preference.  What "more" do we get if we use the SWES account, rather than the free CircleCI option?

**Comment by bonnland** [comment]
@kmpaul 

I ran into Dave Vance in the cafeteria and found out some things.   The NCAR account is used by default for all repos in the NCAR domain.   The main advantage is that we get more CPUs for parallel builds if we want, and we get some extra OS platforms to build on.   Not that useful in this case. 

Dave also said that we can try downloading a CLI for CircleCI that mimics its behavior on a local machine.   It is meant for debugging a configuration, but I wonder if we can hack around with it on a machine with Glade access.  This is totally new territory I'm pretty sure, and I would need time to explore what is possible.

**Comment by kmpaul** [comment]
Wow!  That would be a cool idea to try out CircleCI (or something like it) on a local machine.
